artifacts/vdo-spec.yaml

```diff
@@ -0,0 +1,673 @@
+  namespace: vmware-system-vdo
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: vdo-manager-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: vdo-manager-role
+subjects:
+- kind: ServiceAccount
+  name: vdo-controller-manager
+  namespace: vmware-system-vdo
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: vdo-proxy-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: vdo-proxy-role
+subjects:
+- kind: ServiceAccount
+  name: vdo-controller-manager
+  namespace: vmware-system-vdo
+---
+apiVersion: v1
+data:
+  controller_manager_config.yaml: |
+    apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+    kind: ControllerManagerConfig
+    health:
+      healthProbeBindAddress: :8081
+    metrics:
+      bindAddress: 127.0.0.1:8080
+    webhook:
+      port: 9443
+    leaderElection:
+      leaderElect: true
+      resourceName: 540e4412.vmware.com
+kind: ConfigMap
+metadata:
+  name: vdo-manager-config
+  namespace: vmware-system-vdo
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: vdo-controller-manager-metrics-service
+  namespace: vmware-system-vdo
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    control-plane: controller-manager
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: vdo-controller-manager
+  namespace: vmware-system-vdo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  strategy:
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        control-plane: controller-manager
+    spec:
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=10
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
+        name: kube-rbac-proxy
+        ports:
+        - containerPort: 8443
+          name: https
+      - args:
+        - --health-probe-bind-address=:8081
+        - --metrics-bind-address=127.0.0.1:8080
+        - --leader-elect
+        command:
+        - /manager
+        env:
+        - name: MATRIX_CONFIG_CONTENT
+          valueFrom:
+            configMapKeyRef:
+              key: versionConfigContent
+              name: compat-matrix-config
+              optional: true
+        - name: MATRIX_CONFIG_URL
+          valueFrom:
+            configMapKeyRef:
+              key: versionConfigURL
+              name: compat-matrix-config
+              optional: true
+        - name: AUTO_UPGRADE
+          valueFrom:
+            configMapKeyRef:
+              key: auto-upgrade
+              name: compat-matrix-config
+        image: vmware.com/vdo:40a4440
```

Comment:
This ID is getting used to deploy vdo-controller, however the make deploy makes the docker image with a different tag at runtime, we need to update this spec.yaml with the new tag